workspaces/ui-v2/src/clients/spectacle.ts

```diff
@@ -0,0 +1,90 @@
+import { CQRSCommand } from '../../../optic-domain/build';
+import { IForkableSpectacle } from '../../../spectacle/build';
+
+export class SpectacleClient {
```

Comment:
There's more to pull into spectacle client, but this is a first step since we've been sharing calls across two places

Reply:
I can see how for something as specific as fetching only a single deeply-nested field (commands in this case) creating a client class can make sense, but with how this is named, it seems to suggest all calls to Spectacle must go through the Spectacle client. That would seem to go against one of the major benefits of a GraphQL service: the ability for multiple consumers all to request exactly their own version of a shape they need. 

Not to say that there couldn't be a SpectacleClient (to centralise error handling for example, or for there to be predefined queries), but I'd expect it to be a whole lot more flexible, in line with what GraphQL provides.

Reply:
Hmm yeah, that's definitely not the intent to be prescriptive. We do need to do some work to make the queries have better typing (such that the string you define, spits out the typing you expect). Perhaps error handling, or raising should live in the spectacle instance (in query or mutate) and that better typings are automatically inferred. 

Before we have that though, I think needing to define specific queries and add specific typings is much more important than having the ability for multiple consumers to request their specific data - given most things are through a local http request

Basically, yes I agree, but I think for now as it is implemented this provides more value and we can evolve spectacle to either handle these issues and we remove the need for a spectacle client, or we update the spectacle client to be more graphql friendly.